airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -0,0 +1,214 @@
+    :type gcp_scopes: str
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        connections_path: str,
+        url: Optional[str] = None,
+        auth_type: str = 'token',
+        mount_point: str = 'secret',
+        kv_engine_version: int = 2,
+        token: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        role_id: Optional[str] = None,
+        secret_id: Optional[str] = None,
+        gcp_key_path: Optional[str] = None,
+        gcp_scopes: Optional[str] = None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.connections_path = connections_path.rstrip('/')
+        self.url = url
+        self.auth_type = auth_type
+        self.kwargs = kwargs
+        self.token = token
+        self.username = username
+        self.password = password
+        self.role_id = role_id
+        self.secret_id = secret_id
+        self.mount_point = mount_point
+        self.kv_engine_version = kv_engine_version
+        self.gcp_key_path = gcp_key_path
+        self.gcp_scopes = gcp_scopes
+
+    @cached_property
+    def client(self) -> hvac.Client:
+        """
+        Return an authenticated Hashicorp Vault client
+        """
+
+        _client = hvac.Client(url=self.url, **self.kwargs)
+        if self.auth_type == "token":
+            _client.token = self.token
+        elif self.auth_type == "ldap":
+            _client.auth.ldap.login(
+                username=self.username, password=self.password)
+        elif self.auth_type == "userpass":
+            _client.auth_userpass(username=self.username, password=self.password)
+        elif self.auth_type == "approle":
+            _client.auth_approle(role_id=self.role_id, secret_id=self.secret_id)
+        elif self.auth_type == "github":
+            _client.auth.github.login(token=self.token)
+        elif self.auth_type == "gcp":
+            credentials = self._get_gcp_credentials()
+            _client.auth.gcp.configure(credentials=credentials)
+        else:
+            raise AirflowException(f"Authentication type '{self.auth_type}' not supported")
+
+        if _client.is_authenticated():
+            return _client
+        else:
+            raise VaultError("Vault Authentication Error!")
+
+    def build_path(self, conn_id: str):
+        """
+        Given conn_id, build path for Vault Secret
+
+        :param conn_id: connection id
+        :type conn_id: str
+        """
+        return self.connections_path + "/" + conn_id
+
+    def get_conn_uri(self, conn_id: str) -> Optional[str]:
+        """
+        Get secret value from Vault. Store the secret in the form of URI
+
+        :param conn_id: connection id
+        :type conn_id: str
+        """
+        self.log.debug("Path: %s", self.connections_path)
+        self.log.debug("Mount Point: %s", self.mount_point)
```

Comment:
Do we need to log this everytime (even if it is debug it still seems verbose. to have it on every secret fetch request)

Reply:
Removed